python-package/lightgbm/engine.py

```diff
@@ -276,18 +278,42 @@ class CVBooster:
         The best iteration of fitted model.
     """
 
-    def __init__(self):
+    def __init__(self, model_file=None):
```

Comment:
```suggestion
    def __init__(
        self,
        model_file: Optional[str, pathlib.Path] = None
    ):
```

Per #3756, we are trying to encourage more uses of type hints in the Python package, to improve the ability of type checkers like `mypy` to catch bugs. Can you please add hints here?

Reply:
@jameslamb 
I think that is a great idea. Thanks for the suggestion!